Failure using compose with 1.0.0-SNAPSHOT

Tried setting JDK 11 and 17.

Are there any command line args I need to set for gradle?

Running `./gradlew audio-ui:recordPaparazziDebug`

```
java.lang.reflect.InaccessibleObjectException: Unable to make field private int java.lang.reflect.Field.modifiers accessible: module java.base does not "opens java.lang.reflect" to unnamed module @a9cd3b1
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at app.cash.paparazzi.Paparazzi.forcePlatformSdkVersion(Paparazzi.kt:401)
	at app.cash.paparazzi.Paparazzi.prepare(Paparazzi.kt:138)
	at app.cash.paparazzi.Paparazzi$apply$statement$1.evaluate(Paparazzi.kt:119)
	at app.cash.paparazzi.agent.AgentTestRule$apply$1.evaluate(AgentTestRule.kt:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

A similar one, it is failing with JDK 18

```text
java.lang.InternalError at MethodHandleAccessorFactory.java:167
        Caused by: java.lang.IllegalAccessException at MemberName.java:955
```

This doesn't work

```kotlin
tasks.withType<Test>().configureEach {
    jvmArgs(
        "--add-opens=java.base/java.lang=ALL-UNNAMED",
        "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
        "--add-opens=java.base/java.lang.reflect=ALL-UNNAMED",
    )
}
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,7 +18,7 @@ jobs:
           # Also, see: https://github.com/touchlab/SQLiter/pull/38#issuecomment-867171789
           ubuntu-18.04
         ]
-        java-version: [11, 12, 13, 14, 15]
+        java-version: [11, 12, 16, 18]
 
     runs-on: ${{matrix.os}}
 

--- a/paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt
+++ b/paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt
@@ -74,8 +74,6 @@ import org.junit.rules.TestRule
 import org.junit.runner.Description
 import org.junit.runners.model.Statement
 import java.awt.image.BufferedImage
-import java.lang.reflect.Field
-import java.lang.reflect.Modifier
 import java.util.Date
 import java.util.concurrent.TimeUnit
 import kotlin.coroutines.ContinuationInterceptor
@@ -380,42 +378,15 @@ class Paparazzi @JvmOverloads constructor(
   }
 
   private fun forcePlatformSdkVersion(compileSdkVersion: Int) {
-    val modifiersField =
-      try {
-        Field::class.java.getDeclaredField("modifiers")
-      } catch (e: NoSuchFieldException) {
-        // Hack for Java 12+ access
-        // https://stackoverflow.com/q/56039341
-        // https://github.com/powermock/powermock/commit/66ce9f77215bae68b45f35481abc8b52a5d5b6ae#diff-21c1fc51058efd316026f11f34f51c5c
-        try {
-          val getDeclaredFields0 =
-            Class::class.java.getDeclaredMethod(
-              "getDeclaredFields0",
-              Boolean::class.javaPrimitiveType
-            )
-          getDeclaredFields0.isAccessible = true
-          val fields = getDeclaredFields0.invoke(Field::class.java, false) as Array<Field>
-          fields.find { it.name == "modifiers" } ?: throw e
-        } catch (ex: Exception) {
-          e.addSuppressed(ex)
-          throw e
-        }
-      }
-    modifiersField.isAccessible = true
-
-    val versionClass = try {
+    val buildVersionClass = try {
       Paparazzi::class.java.classLoader.loadClass("android.os.Build\$VERSION")
     } catch (e: ClassNotFoundException) {
+      // Project unit tests don't load Android platform code
       return
     }
-
-    versionClass
-      .getDeclaredField("SDK_INT")
-      .apply {
-        isAccessible = true
-        modifiersField.setInt(this, modifiers and Modifier.FINAL.inv())
-        setInt(null, compileSdkVersion)
-      }
+    buildVersionClass
+      .getFieldReflectively("SDK_INT")
+      .setStaticValue(compileSdkVersion)
   }
 
   private fun initializeAppCompatIfPresent() {

--- a/paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Reflections.kt
+++ b/paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Reflections.kt
@@ -1,0 +1,62 @@
+package app.cash.paparazzi
+
+import sun.misc.Unsafe
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+import java.security.AccessController
+import java.security.PrivilegedAction
+
+/**
+ * Inspired by and ported from:
+ * https://github.com/powermock/powermock/commit/fc092c5d7e339d01e079184a2a0e88b5c46fc0e8
+ * https://github.com/powermock/powermock/commit/bd92bcc5329c4981cf09dece5c3eafcf92fe49ff
+ */
+internal fun Class<*>.getFieldReflectively(fieldName: String): Field =
+  try {
+    this.getDeclaredField(fieldName).also { it.isAccessible = true }
+  } catch (e: NoSuchFieldException) {
+    throw RuntimeException("Field '$fieldName' was not found in class $name.")
+  }
+
+internal fun Field.setStaticValue(value: Any) {
+  try {
+    this.isAccessible = true
+    val isFinalModifierPresent = this.modifiers and Modifier.FINAL == Modifier.FINAL
+    if (isFinalModifierPresent) {
+      AccessController.doPrivileged<Any?>(
+        PrivilegedAction {
+          try {
+            val unsafe = Unsafe::class.java.getFieldReflectively("theUnsafe").get(null) as Unsafe
+            val offset = unsafe.staticFieldOffset(this)
+            val base = unsafe.staticFieldBase(this)
+            unsafe.setFieldValue(this, base, offset, value)
+            null
+          } catch (t: Throwable) {
+            throw RuntimeException(t)
+          }
+        }
+      )
+    } else {
+      this.set(null, value)
+    }
+  } catch (ex: SecurityException) {
+    throw RuntimeException(ex)
+  } catch (ex: IllegalAccessException) {
+    throw RuntimeException(ex)
+  } catch (ex: IllegalArgumentException) {
+    throw RuntimeException(ex)
+  }
+}
+
+internal fun Unsafe.setFieldValue(field: Field, base: Any, offset: Long, value: Any) =
+  when (field.type) {
+    Integer.TYPE -> this.putInt(base, offset, (value as Int))
+    java.lang.Short.TYPE -> this.putShort(base, offset, (value as Short))
+    java.lang.Long.TYPE -> this.putLong(base, offset, (value as Long))
+    java.lang.Byte.TYPE -> this.putByte(base, offset, (value as Byte))
+    java.lang.Boolean.TYPE -> this.putBoolean(base, offset, (value as Boolean))
+    java.lang.Float.TYPE -> this.putFloat(base, offset, (value as Float))
+    java.lang.Double.TYPE -> this.putDouble(base, offset, (value as Double))
+    Character.TYPE -> this.putChar(base, offset, (value as Char))
+    else -> this.putObject(base, offset, value)
+  }

